talentsearch/resources/js/components/skills/SkillChecklist.tsx

```diff
@@ -0,0 +1,152 @@
+import React from "react";
+import groupBy from "lodash/groupBy";
+import sum from "lodash/sum";
+import { useIntl } from "react-intl";
+import { getLocale } from "@common/helpers/localize";
+import { SkillFamily } from "@common/api/generated";
+import {
+  getSkillCategoryText,
+  getSkillCategoryImage,
+} from "@common/constants/localizedConstants";
+
+interface FamilyProps {
+  family: SkillFamily;
+  checked: boolean;
+  callback: (family: SkillFamily, checked: boolean) => void;
+}
+
+const Family: React.FunctionComponent<FamilyProps> = ({
+  family,
+  checked,
+  callback,
+}) => {
+  const intl = useIntl();
+  const locale = getLocale(intl);
+
+  const uncheckedStyle = { "data-h2-font-weight": "b(400)" };
+  const checkedStyle = { "data-h2-font-weight": "b(700)" };
+
+  return (
+    <div
+      {...(checked ? checkedStyle : uncheckedStyle)}
+      data-h2-padding="b(all, xxs)"
+      key={family.key}
+    >
+      <input
+        type="checkbox"
+        id={family.id}
+        onChange={(e) => {
+          callback(family, e.target.checked);
+        }}
+      />
+      &nbsp;
+      <label htmlFor={family.id}>
+        {family.name?.[locale]} ({family.skills ? family.skills.length : 0})
+      </label>
+    </div>
+  );
+};
+
+interface CategoryProps {
+  category: string;
+  skillFamilies: SkillFamily[];
+  callback: (selected: SkillFamily, added: boolean) => void;
+}
+
+const Category: React.FunctionComponent<CategoryProps> = ({
+  category,
+  skillFamilies,
+  callback,
+}) => {
+  const intl = useIntl();
+
+  // The app currently crashes if no description is specified for the category.
+  // Do we want to avoid this?
+  const categoryName = intl.formatMessage(getSkillCategoryText(category));
+  const skillCount = sum(
+    skillFamilies.map((family) => (family.skills ? family.skills.length : 0)),
+  );
+  const title = `${categoryName} (${skillCount})`;
+  const image = getSkillCategoryImage(category);
+
+  const [selectedFamilies, setSelectedFamilies] = React.useState<SkillFamily[]>(
+    [],
+  );
+
+  const handleChecked = (family: SkillFamily, checked: boolean) => {
+    if (checked) {
+      setSelectedFamilies([...selectedFamilies, family]);
+    } else {
+      setSelectedFamilies(selectedFamilies.filter((elem) => elem !== family));
```

Comment:
Always avoid comparing non-primitive objects, if possible, because they could hold identical data while not being equal. If an id or key is available, use that.
```suggestion
      setSelectedFamilies(selectedFamilies.filter((elem) => elem.id !== family.id));
```